b9lab-content/3-main-concepts/04-accounts.md

```diff
@@ -1 +1,141 @@
+
+A Hyper-deterministic Wallet uses a single seed phrase to generate many key pairs. In this way, only the seed phrase needs to be backed up.
+
+Cosmos SDK uses [BIP32](https://en.bitcoin.it/wiki/BIP_0032) which allows users to generate a set of accounts from an initial secret that usually consists of 12 or 24 words known as the mnemonic. Importantly, key pairs can always be reproduced from the mnemonic.
+
+To see BIP32 in action, visit [https://www.bip32.net/](https://www.bip32.net/). Click `Show entropy details` and enter random data in the `Entropy` field. This reveals an important aspect of the initial seed generation process. A source of randomness is essential. As you provide entropy, the `BIP39 Mnemonic` field will begin to populate with words. Scrolling down further and selecting the `BIP39` tab under `Derived Addresses`, you will see `Public Key` and `Private Key` pairs.
+
+Like most blockchain implementations, Cosmos derives addresses from the public keys. More on this below.
+
+![hd accounts](./images/hd-accounts.png)
+
+Using BIP39, a user is required only to store their BIP39 mnemonic in a safe and confidential manner. All key pairs can be reconstructed from the mnemonic because it’s deterministic. There is no practical upper limit to the number of key pairs that can be generated from a single mnemonic, hence the name hyper-deterministic that is used to describe this approach to key generation.
+
+## Keyring, Addresses and Address Types
+
+In the Cosmos SDK, keys are stored and managed in an object called a Keyring.
+
+Authentication is implemented as signature verification.
+
+* Users generate transactions, sign transactions and send the signed transaction to the blockchain.
+* Transactions are formatted with the public key as part of the message. Signatures are verified by confirming that the signature’s public key matches the public key in the message. If the message is signed by anyone other than the purported sender, then the signature is invalid and the transaction is rejected.
+
+In case the foregoing is unclear, consider the following pseudo message:
+
+```
+Message: {
+      Payload: {
+  Sender: “0x1234”,
+  Data: “Hello World”}
+      Signature: “0xabcd”
+}
+```
+Passing `Payload` into the signature verification function will return a sender. The derived sender must match the sender in the `Payload` itself. This will confirm that the `Payload` could only originate from someone with knowledge of the private key corresponding to `Sender: “0x1234”`
+
+## Signature Schemes
+
+As you might guess, there is more than one implementation of the public key signature process described so far. Cosmos SDK supports the following digital key schemes for creating digital signatures:
+
+* [secp256k1](https://www.secg.org/sec2-v2.pdf), as implemented in the SDK's crypto/keys/secp256k1 package.
```

Comment:
Should probably mention that most use secp256k1 and this is the same as Bitcoin ?